SIS_import/sis_set_url.rb

```diff
@@ -29,7 +46,8 @@
 def sleep_according_to_timer_and_api_call_limit(start_time, end_time, call_count, time_interval_in_seconds, allowed_call_number_during_interval)
 	# if meet max allowed call count during the time interval
 	# sleep until time expires
-	while (Time.now.to_i <  end_time.to_i && call_count > allowed_call_number_during_interval)
+	while (Time.now.to_i <=  end_time.to_i && call_count >= allowed_call_number_during_interval)
+		p "sleep till next time interval"
 		sleep(1)
```

Comment:
sleep(1) is for sleep one second. It may be appropriate for ESB wait and maybe too short for Canvas call...